conf/script/prov-m0-reset

```diff
@@ -244,137 +284,214 @@ get_fid() {
 get_proc_state() {
     fid=$1
 
-    state=$(/opt/seagate/cortx/hare/bin/consul kv get processes/$fid 2> /dev/null | jq -r '.state')
+    state=$(consul kv get processes/$fid 2> /dev/null | jq -r '.state')
     echo $state
 }
 
+run_cmd() {
+    local node=$1; shift
+    local cmd=$*
+    ssh $node $cmd
+}
+
+get_leader() {
+    consul kv get leader
+}
+
+cleanup() {
+    log "${FUNCNAME[0]}: Stopping Hax.."
+    run_cmd $node2 "sudo systemctl stop hare-hax-c2"
+    run_cmd $node1 "sudo systemctl stop hare-hax-c1"
+
+    log "${FUNCNAME[0]}: Stopping Consul agents.."
+    run_cmd $node2 "sudo systemctl stop hare-consul-agent-c2"
+    run_cmd $node1 "sudo systemctl stop hare-consul-agent-c1"
+
+    log "${FUNCNAME[0]}: Deleting lnet interfaces.."
+    run_cmd $node1 "[ `sudo lctl list_nids | grep $ip1` ] && sudo lnetctl net del --net $net_type --if $left_iface:c1" || true
+    run_cmd $node2 "[ `sudo lctl list_nids | grep $ip2` ] && sudo lnetctl net del --net $net_type --if $right_iface:c2" || true
+
+    log "${FUNCNAME[0]}: Restarting motr-kernel.."
+    run_cmd $node1 "sudo systemctl restart motr-kernel"
+    run_cmd $node2 "sudo systemctl restart motr-kernel"
+
+    run_cmd $node1 "sudo systemctl daemon-reload"
+    run_cmd $node2 "sudo systemctl reset-failed"
+
+    log "${FUNCNAME[0]}: Removing ipaddresses.."
+    run_cmd $node1 "ip addr del $ip1/24 dev $left_iface:c1 || true"
+    run_cmd $node2 "ip addr del $ip2/24 dev $right_iface:c1 || true"
+
+    log "${FUNCNAME[0]}: Unmounting /var/motr devices.."
+    run_cmd $node1 "mountpoint -q /var/motr1 && sudo umount /var/motr1 || true"
+    run_cmd $node2 "mountpoint -q /var/motr2 && sudo umount /var/motr2 || true"
+}
+
 m0_mkfs() {
+    local node1=$lnode
+    local node2=$rnode
+    if $single_node; then
+        node1=$(/opt/seagate/cortx/hare/libexec/node-name)
+        node2=$(/opt/seagate/cortx/hare/libexec/node-name)
+    fi
 
     local confd_fid_c1=$(get_fid 'confd' $lnode)
     local ios_fid_c1=$(get_fid 'ios' $lnode)
 
     local confd_fid_c2=$(get_fid 'confd' $rnode)
     local ios_fid_c2=$(get_fid 'ios' $rnode)
 
-    systemctl is-active --quiet m0d@$confd_fid_c1 ||
-    systemctl is-active --quiet m0d@$ios_fid_c1 ||
-    systemctl is-active --quiet m0d@$confd_fid_c2 ||
-    systemctl is-active --quiet m0d@$ios_fid_c2 &&
+    run_cmd $node1 "systemctl is-active --quiet m0d@$confd_fid_c1" ||
+    run_cmd $node1 "systemctl is-active --quiet m0d@$ios_fid_c1" ||
+    run_cmd $node2 "systemctl is-active --quiet m0d@$confd_fid_c2" ||
+    run_cmd $node2 "systemctl is-active --quiet m0d@$ios_fid_c2" &&
         die 'Active m0d instances found, please stop all the m0d instances'
 
-    sudo systemctl start motr-mkfs@$confd_fid_c1
+    run_cmd $node1 "sudo systemctl start motr-mkfs@$confd_fid_c1"
     while [[ $(get_proc_state $confd_fid_c1) != M0_CONF_HA_PROCESS_STOPPED ]]; do
```

Comment:
What is there is a consul failure and consul is not accessible, how it is being handled ?

Reply:
We make sure that consul has started successfully in mkfs() function. We are waiting for its leader to be elected and only then proceed with starting further services. If Consul fails here then its need to be addressed separately.